enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+traditional relational database. In order to integrate with OpenShift
+infrastructure management tooling, it is necessary to additionally expose its
+capabilities as Kubernetes-native APIs. They will be implemented as Custom
+Resource Definitions and a controller with the assisted service's capabilities.
+
+For the first deliverable, a local deployment of the backing service will
+include new controllers that will operate to achieve the desired state as
+declared in CRDs. Thus the service will expose both REST and Kubernetes-native
+APIs, and both API layers will call the same set of internal methods. The local
+service will include a database; sqlite if possible, or else postgresql.
+
+    --------------------------
+    |  REST API  |  k8s API  |
+    --------------------------
+    | Assisted Service logic |
+    --------------------------
+        |               |
+        v               v
+      SQL DB        File System
+                      storage
+
+For modest clusters that do not have persistent storage available but need
+day-2 ability to add nodes, the database will need to be reconstructable in
+case its current state is lost. Thus the CRs will be the source of truth for
+the specs, and the controller will upon statup first ensure that the DB
+reflects the desired state in CRs. The source of truth for the status, however,
+is the actual state of the agents running on the hosts being installed and not
+necessarily what was previously recorded.
+
+Agents that are running on hardware and waiting for installation-related
+instructions will continue to communicate with the assisted service using the
+existing REST API. In the future that may transition to using a CRD.
+
+**AgentDrivenInstallationImage**
+This resource represents a discovery image that should be used for booting
+hosts. In the REST API this resource is embedded in the "cluster" resource, but
+for kubernetes-native is more natural for it to be separate.
+
+Spec:
+ * clusterName: string
+ * sshPublicKey: string
+ * pullSecretName: string (reference to k8s secret)
+ * ignitionOverrides: object
+ * staticIpConfiguration: object
+
+Status:
+ * state: string
+ * sizeBytes: integer
+ * downloadUrl: string
+ * createdAt: string
+ * expiresAt: string
+
+**AgentDrivenClusterInstallation**
+This resource correlates to the "cluster" resource in the assisted installer's
+current API.
+
+Spec:
+ * name: string
+ * approved: boolean (used for starting installation)
+ * openshiftVersion: string
+ * baseDnsDomain: string
+ * clusterNetworkCidr: string
+ * clusterNetworkHostPrefix: integer
+ * serviceNetworkCidr: string
+ * apiVip: string
+ * apiVipDnsName: string
+ * ingressVip: string
+ * machineNetworkCidr: string
+ * sshPublicKey: string
+ * vipDhcpAllocation: boolean
+ * httpProxy: string
+ * httpsProxy: string
+ * noProxy: string
+ * userManagedNetworking: boolean
+ * additionalNtpSource: string
+ * installConfigOverrides: string
+
+Status:
```

Comment:
Suggest taking a look through https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties for the standard use of Conditions, and then trying to apply it to this status here. It would go a long way to making this appear as a standard kube API.